pkg/upgrade/util.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2022 NVIDIA CORPORATION & AFFILIATES
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/Mellanox/network-operator/pkg/consts"
+)
+
+func ChangeNodeUpgradeState(
+	node *v1.Node, newNodeState string, k8sClient client.Client, log logr.Logger) error {
+	log.V(consts.LogLevelInfo).Info("Updating node state",
+		"node", node.Name,
+		"new state:", newNodeState)
+
+	node.SetManagedFields(nil)
```

Comment:
Why we need to remove managed fields? Removing of the content of this field can be problematic, because this can break node object update logic in other controllers.

Reply:
I've added it to reduce the rate at which "Operation cannot be fulfilled on nodes the object has been modified" occurs but now I've changed the solution to re-requesting the object and trying again if it happens. So `node.SetManagedFields(nil)` code is removed